cmd/flakechecker/main_test.go

```diff
@@ -0,0 +1,159 @@
+			bqClient: &biqQueryClientMock{
+				getRatioForTest: func(_ *flakeCheckerRecord, _ string) (int, int, error) {
+					return 0, 0, errors.New("fail")
+				},
+			},
+			failedTests: []testcase.TestCase{{Name: "test", Classname: "class"}},
+			flakeConfigs: []*flakeCheckerRecord{
+				{MatchJobName: "test-job", TestNameRegex: "test", Classname: "class", RatioThreshold: 1},
+			},
+			expectError: true,
+		},
+		"total runs below limit": {
+			bqClient: &biqQueryClientMock{
+				getRatioForTest: func(_ *flakeCheckerRecord, _ string) (int, int, error) {
+					return totalRunsLimit - 1, 0, nil
+				},
+			},
+			failedTests: []testcase.TestCase{{Name: "test", Classname: "class"}},
+			flakeConfigs: []*flakeCheckerRecord{
+				{MatchJobName: "test-job", TestNameRegex: "test", Classname: "class", RatioThreshold: 1},
+			},
+			expectError: true,
+		},
+		"fail ratio below threshold": {
+			bqClient: &biqQueryClientMock{
+				getRatioForTest: getRatioForTestNoFailures,
+			},
+			failedTests: []testcase.TestCase{{Name: "test", Classname: "class"}},
+			flakeConfigs: []*flakeCheckerRecord{
+				{MatchJobName: "test-job", TestNameRegex: "test", Classname: "class", RatioThreshold: 1},
+			},
+			expectError: false,
+		},
+		"fail ratio above threshold": {
+			bqClient: &biqQueryClientMock{
+				getRatioForTest: getRatioForTestAllFailures,
+			},
+			failedTests: []testcase.TestCase{{Name: "test", Classname: "class"}},
+			flakeConfigs: []*flakeCheckerRecord{
+				{MatchJobName: "test-job", TestNameRegex: "test", Classname: "class", RatioThreshold: 1},
+			},
+			expectError: true,
+		},
+		"fail ratio below threshold - multiple tests": {
+			bqClient: &biqQueryClientMock{
+				getRatioForTest: getRatioForTestNoFailures,
+			},
+			failedTests: []testcase.TestCase{
+				{Name: "test", Classname: "class"},
+				{Name: "test-1", Classname: "class-1"},
+			},
+			flakeConfigs: []*flakeCheckerRecord{
+				{MatchJobName: "test-job", TestNameRegex: "test", Classname: "class", RatioThreshold: 1},
+				{MatchJobName: "test-job", TestNameRegex: "test-1", Classname: "class-1", RatioThreshold: 1},
+			},
+			expectError: false,
+		},
+		"fail ratio above threshold - multiple tests": {
+			bqClient: &biqQueryClientMock{
+				getRatioForTest: getRatioForTestAllFailures,
+			},
+			failedTests: []testcase.TestCase{
+				{Name: "test-ratio-below", Classname: "class"},
+				{Name: "test-ratio-above", Classname: "class"},
+			},
+			flakeConfigs: []*flakeCheckerRecord{
+				{MatchJobName: "test-job", TestNameRegex: "test-ratio-below", Classname: "class", RatioThreshold: 90},
+				{MatchJobName: "test-job", TestNameRegex: "test-ratio-above", Classname: "class", RatioThreshold: 10},
+			},
+			expectError: true,
+		},
+	}
+
+	for sampleName, sample := range samples {
+		t.Run(sampleName, func(tt *testing.T) {
+			err := p.checkFailedTests(sample.bqClient, sample.failedTests, sample.flakeConfigs)
+
+			assert.Equal(tt, sample.expectError, err != nil)
```

Comment:
I'm not a fan of binary error checking as error may occur for different reason then we expected